devices/uart-stm32l4x6/uart.c

```diff
@@ -0,0 +1,255 @@
+/*
+ * Phoenix-RTOS
+ *
+ * plo - operating system loader
+ *
+ * STM32L4x6 Serial driver
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Aleksander Kaminski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include <lib/errno.h>
+#include <lib/lib.h>
+#include <devices/devs.h>
+
+
+typedef struct {
+	volatile unsigned int *base;
+	volatile unsigned char rxdfifo[128];
+	volatile unsigned int rxdr;
+	volatile unsigned int rxdw;
+	volatile int rxflag;
+} uart_t;
+
+
+struct {
+	uart_t uarts[UART_MAX_CNT];
+} uart_common;
+
+
+enum { cr1 = 0, cr2, cr3, brr, gtpr, rtor, rqr, isr, icr, rdr, tdr };
+
+
+static int uartLut[] = { UART1, UART2, UART3, UART4, UART5 };
+
+
+static const struct {
+	volatile unsigned int *base;
+	int dev;
+	unsigned int irq;
+	int rxport;
+	unsigned char rxpin;
+	unsigned char rxaf;
+	int txport;
+	unsigned char txpin;
+	unsigned char txaf;
+} uartInfo[] = {
+	{ (void *)0x40013800, pctl_usart1, usart1_irq, pctl_gpioa, 10, 7, pctl_gpioa, 9, 7 },
```

Comment:
Why don't you use `UART*_BASE` and `UART*_CLOCK` from peripherals?

Reply:
Done